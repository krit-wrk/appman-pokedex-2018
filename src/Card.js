
import React from 'react'
import { Row, Col, Box } from './Layout'

export function Card(props) {
    const c = props.card
    return <div>
        <Row>

            <img width="100px" src={c.imageUrl} />
            <Box>
                <Col>
                    <label>  {c.name}</label>
                    <label>HP:{calHP(c)}</label>
                    <label>Strength:{calStrength(c)}</label>
                    <label>Weakness:{calWeakness(c)}</label>
                    {/* <label>calDamage:{calDamage(c)}</label> */}
                    <label>Happiness: {calHappiness(c)}</label>
                    <button onClick={_ => props.action(c)}>{props.btnDisplay}</button>
                </Col>
            </Box>
        </Row>


    </div>
}

function calHP(card) {
    const { hp } = card
    if (hp >= 100) return 100;
    if (hp <= 0) return 100;
    if (hp <= 100 && hp >= 0)
        return hp
    return 0
}

function calStrength(card) {
    const { attacks } = card
    var length = 0
    try {
        length = attacks.length
    } catch{ }

    if (length == 2) return 100;
    if (length == 1) return 50;
    return 0
}
function calWeakness(card) {
    const { weaknesses } = card

    var length = 0
    try {
        length = weaknesses.length
    } catch{ }


    if (length == 1) return 100;
    return 0
}
function calDamage(card) {
    const { attacks } = card
    var sum = 0
    if (Array.isArray(attacks))
        attacks.forEach(a => {
            var d = a.damage ? a.damage : 0
            sum += Math.abs(Number.parseInt(d))
        })
    return sum
}

function calHappiness(card) {
    return ((calHP(card) / 10) + (calDamage(card) / 10) + 10 - (calWeakness(card))) / 5
}
