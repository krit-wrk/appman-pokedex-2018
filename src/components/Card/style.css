.component-card {
  box-shadow: 1px 1px 10px 1px #d5d6dc;
}
.component-card:hover {
  box-shadow: 1px 1px 10px 1px #aeaeae;
}
