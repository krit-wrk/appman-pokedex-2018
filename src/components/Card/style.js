import { COLORS } from "./../../constants";
import { StyleSheet } from "aphrodite";
export default StyleSheet.create({
	container: {
		display: "flex",
		flexDirection: "row",
		flex: 1,
		padding: "8px",
		backgroundColor: COLORS.CardBackground,
		boxShadow: `1px 1px 10px 1px ${COLORS.CardShadow}`,
		":hover": {
			boxShadow: `1px 1px 10px 1px ${COLORS.CardShadowHover}`,
		},
	},

	image: {
		width: "150px",
	},
	desc: {
		flex: 1,
		flexDirection: "column",
		padding: "8px",
	},

	name: {
		fontSize: "40px",
		fontFamily: "Gaegu",
	},

	descContainer: {
		display: "flex",
		flexDirection: "row",
		flex: 1,
		alignItems: "center",
	},
	descText: {
		fontSize: "25px",
		width: "100px",
	},
	descBar: {
		display: "flex",
		flex: 1,
		maxWidth: "50%",
	},

	cuteContainer: {
		display: "flex",
		flexDirection: "row",
		flex: 1,
		alignItems: "center",
	},
	cuteImage: {
		width: "50px",
		padding: "2px",
	},
});
