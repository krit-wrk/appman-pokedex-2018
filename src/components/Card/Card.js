import React, { useMemo } from 'react'
import style from './style.js'
import { css } from 'aphrodite'
import { ProgressBar } from './..'
import cute from './../../assets/cute.png';
const Card = ({ image, name, hp, str, weak, happy, btn }) => {

    const happyList = useMemo(() => {
        const list = []
        for (let i = 1; i <= happy; i++) {
            list.push(<img key={i} className={css(style.cuteImage)} src={cute} alt="cute" />)
        }
        return list
    }, [happy])

    return (
        <div className={css(style.container)} >
            <img className={css(style.image)} src={image} alt={name} />
            <div className={css(style.desc)}>
                <div className={css(style.name)}>{name.toUpperCase()}</div>
                <div className={css(style.descContainer)}>
                    <div className={css(style.descText)}>HP</div>
                    <div className={css(style.descBar)}><ProgressBar value={hp} /></div>
                </div>
                <div className={css(style.descContainer)}>
                    <div className={css(style.descText)}>STR</div>
                    <div className={css(style.descBar)}><ProgressBar value={str} /></div>
                </div>
                <div className={css(style.descContainer)}>
                    <div className={css(style.descText)}>WEAK</div>
                    <div className={css(style.descBar)}><ProgressBar value={weak} /></div>
                </div>
                <div className={css(style.cuteContainer)}>
                    {happyList}
                </div>

            </div >
        </div >
    )
}
export { Card }