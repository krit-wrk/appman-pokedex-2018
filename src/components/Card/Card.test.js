import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Card } from "./Card";
import { StyleSheetTestUtils } from "aphrodite";
describe("Card", () => {
	beforeEach(() => {
		StyleSheetTestUtils.suppressStyleInjection();
	});
	afterEach(() => {
		StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
	});
	it("renders without crashing", () => {
		const { getByText, container } = render(
			<Card image="1" name="2" hp="3" str="4" weak="5" happy="6" btn="X" />,
		);
		expect(container).toMatchSnapshot();
		fireEvent.mouseEnter(getByText("2"));
		const x = getByText("X");
		expect(x).toBeInTheDocument();
		expect(container).toMatchSnapshot();
		fireEvent.mouseLeave(getByText("2"));
		expect(x).not.toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
