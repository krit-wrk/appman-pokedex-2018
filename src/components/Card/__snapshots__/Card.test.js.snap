// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without crashing 1`] = `
<div>
  <div
    class="container_b4ahsr"
  >
    <div
      class="imageContainer_1g6vvub"
    >
      <img
        alt="2"
        class="image_5hddva"
        src="1"
      />
    </div>
    <div
      class="desc_19c633u"
    >
      <div
        class="titleContainer_1ryugy2"
      >
        <div
          class="name_4p276u"
        >
          2
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          HP
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1jumns9"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          STR
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1so9ooe"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          WEAK
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_3eix2n"
            />
          </div>
        </div>
      </div>
      <div
        class="cuteContainer_zz3svu"
      >
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Card renders without crashing 2`] = `
<div>
  <div
    class="container_b4ahsr"
  >
    <div
      class="imageContainer_1g6vvub"
    >
      <img
        alt="2"
        class="image_5hddva"
        src="1"
      />
    </div>
    <div
      class="desc_19c633u"
    >
      <div
        class="titleContainer_1ryugy2"
      >
        <div
          class="name_4p276u"
        >
          2
        </div>
        <div>
          X
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          HP
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1jumns9"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          STR
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1so9ooe"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          WEAK
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_3eix2n"
            />
          </div>
        </div>
      </div>
      <div
        class="cuteContainer_zz3svu"
      >
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Card renders without crashing 3`] = `
<div>
  <div
    class="container_b4ahsr"
  >
    <div
      class="imageContainer_1g6vvub"
    >
      <img
        alt="2"
        class="image_5hddva"
        src="1"
      />
    </div>
    <div
      class="desc_19c633u"
    >
      <div
        class="titleContainer_1ryugy2"
      >
        <div
          class="name_4p276u"
        >
          2
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          HP
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1jumns9"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          STR
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_1so9ooe"
            />
          </div>
        </div>
      </div>
      <div
        class="descContainer_14nb90c"
      >
        <div
          class="descText_1r7w1yk"
        >
          WEAK
        </div>
        <div
          class="descBar_2jj7wg"
        >
          <div
            class="container_tze1ke"
          >
            <div
              class="bar_1o8e29j-o_O-bar_3eix2n"
            />
          </div>
        </div>
      </div>
      <div
        class="cuteContainer_zz3svu"
      >
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
        <img
          alt="cute"
          class="cuteImage_1pmj90s"
          src="cute.png"
        />
      </div>
    </div>
  </div>
</div>
`;
